utils/settings.js

```diff
@@ -1,94 +1,74 @@
-const db = require("./database.js");
+const db = require('./database.js');
 const settings = {};
 
 const keyWords = {
-	add: ["add", "push"],
-	remove: ["remove", "delete"],
-	true: ["1","on","true"],
-	false: ["0","off","false"]
+	true: ['1', 'on', 'true'],
+	false: ['0', 'off', 'false'],
 };
 
-
-const boolParser = input => {
+const boolParser = (input) => {
 	if (keyWords.true.includes(input)) return 1;
 	if (keyWords.false.includes(input)) return 0;
 	throw `Invalid argument \`${input}\`, argument should be of type: \`bool\`.`;
 };
 
-const idChecker = (collection, id) => {
-	if (!collection.get(id)) throw `Could not find \`${id}\` on this server.`;
-};
-
-const modifyArray = (flag, array, item, collection) => {
-	let itemArray = [item];
-	if (Array.isArray(item)) itemArray = item.slice();
-	let adding;
-	itemArray.forEach(item => {
-		const itemExists = array.includes(item);
-		if (!itemExists) idChecker(collection, item);
-		
-		if (keyWords.add.includes(flag)) {
-			adding = true;
-			if (itemExists) throw `\`${item}\` already listed.`;
-		} else if (keyWords.remove.includes(flag)){
-			if (!itemExists) throw `\`${item}\` not listed.`;
-			adding = false;
+const collectionContainsId = (collection, id) => collection.has(id);
+
+const mutateArrayOfIds = (arrayToModify, itemArray, validIdCollection) => {
+	itemArray.forEach((item) => {
+		const itemExistsInArray = arrayToModify.includes(item);
+		if (!itemExistsInArray) {
+			if (collectionContainsId(validIdCollection, item)) {
+				// valid id supplied, add item into arrayToModify
+				arrayToModify.push(item);
+			} else {
+				// invalid id supplied, throw err
+				throw `Could not find \`${item}\` on this server.`;
+			}
 		} else {
-			throw `Unknown key word: \`${flag}\`.`; 
+			// item does exist, remove it
+			let index = arrayToModify.indexOf(item);
+			arrayToModify.splice(index, 1);
 		}
 	});
-
-	if (adding) {
-		array.push(...itemArray);
-	} else {
-		itemArray.forEach(item => {
-			let index = array.indexOf(item);
-			array.splice(index, 1);
-		});
-	}
 };
 
-const roleListHandler = (id, input, key, collection) => {
-	const roles = (settings[id][key] || []).slice();
-	if (input.length === 1) {
-		modifyArray("add", roles, input[0], collection);
-		return roles;
-	}
-	modifyArray(input[0], roles, input.slice(1), collection);
+const roleListHandler = (guildId, input, settingName, validIdCollection) => {
+	const roles = (settings[guildId][settingName] || '').slice();
+	mutateArrayOfIds(roles, input, validIdCollection);
 	if (roles.length === 0) return null;
 	return roles;
 };
 
 // settings validators
 const validSettings = {
```

Comment:
These are no longer settingValidators because they transform and return the new setting value for the database, instead of just validating